src/core_write.cpp

```diff
@@ -334,7 +334,10 @@ void TxToUniv(const CTransaction& tx, const uint256& hashBlock, bool include_add
     }
 
     if (calculate_fee) {
-        const CAmount fee = amt_total_in - amt_total_out;
+        CAmount fee = amt_total_in - amt_total_out;
+        if (tx.IsPlatformTransfer()) {
+            fee = CHECK_NONFATAL(GetTxPayload<CAssetUnlockPayload>(tx))->getFee();
```

Comment:
@UdjinM6 

```suggestion
            const auto optAssetUnlock{GetTxPayload<CAssetUnlockPayload>(tx)};
            CHECK_NONFATAL(optAssetUnlock);
            fee = ->getFee(optAssetUnlock);
```

Reply:
maybe include changes in `src/util/check.h` from 10203560f5367acd9cd96b4857b8b4a223c73550 ?